stl/inc/algorithm

```diff
@@ -2214,6 +2217,163 @@ namespace ranges {
+            requires (forward_iterator<_It1> || sized_sentinel_for<_Se1, _It1>)
+                  && (forward_iterator<_It2> || sized_sentinel_for<_Se2, _It2>)
+                  && indirectly_comparable<_It1, _It2, _Pr, _Pj1, _Pj2>
+        _NODISCARD constexpr bool operator()(_It1 _First1, _Se1 _Last1, _It2 _First2, _Se2 _Last2, _Pr _Pred = {},
+            _Pj1 _Proj1 = {}, _Pj2 _Proj2 = {}) const {
+            // clang-format on
+            _Adl_verify_range(_First1, _Last1);
+            _Adl_verify_range(_First2, _Last2);
+
+            auto _UFirst1 = _Get_unwrapped(_STD move(_First1));
+            auto _ULast1  = _Get_unwrapped(_STD move(_Last1));
+            auto _UFirst2 = _Get_unwrapped(_STD move(_First2));
+            auto _ULast2  = _Get_unwrapped(_STD move(_Last2));
+            if constexpr (same_as<_Se1, unreachable_sentinel_t> && same_as<_Se2, unreachable_sentinel_t>) {
+                return _Equal_4(_STD move(_UFirst1), _STD move(_ULast1), _STD move(_UFirst2), _STD move(_ULast2),
+                    _Pass_fn(_Pred), _Pass_fn(_Proj1), _Pass_fn(_Proj2));
+            } else if constexpr (same_as<_Se1, unreachable_sentinel_t> || same_as<_Se2, unreachable_sentinel_t>) {
+                return false;
+            } else {
+                iter_difference_t<_It1> _Count1;
+                if constexpr (sized_sentinel_for<_Se1, _It1>) { // TRANSITION, LWG-3392
+                    _Count1 = _ULast1 - _UFirst1;
+                } else {
+                    _Count1 = _RANGES distance(_UFirst1, _ULast1);
+                }
+
+                iter_difference_t<_It2> _Count2;
+                if constexpr (sized_sentinel_for<_Se2, _It2>) { // TRANSITION, LWG-3392
+                    _Count2 = _ULast2 - _UFirst2;
+                } else {
+                    _Count2 = _RANGES distance(_UFirst2, _ULast2);
+                }
+
+                if (_Count2 > _Count1) {
+                    return false;
+                }
+
+                _RANGES advance(_UFirst1, static_cast<iter_difference_t<_It1>>(_Count1 - _Count2));
+                return _RANGES _Equal_count(_STD move(_UFirst1), _STD move(_UFirst2),
+                    static_cast<iter_difference_t<_It1>>(_Count2), _Pass_fn(_Pred), _Pass_fn(_Proj1), _Pass_fn(_Proj2));
+            }
+        }
+
+        // clang-format off
+        template <input_range _Rng1, input_range _Rng2, class _Pr = ranges::equal_to, class _Pj1 = identity,
+            class _Pj2 = identity>
+            requires (forward_range<_Rng1> || sized_range<_Rng1>) && (forward_range<_Rng2> || sized_range<_Rng2>)
+                  && indirectly_comparable<iterator_t<_Rng1>, iterator_t<_Rng2>, _Pr, _Pj1, _Pj2>
+        _NODISCARD constexpr bool operator()(
+                _Rng1&& _Range1, _Rng2&& _Range2, _Pr _Pred = {}, _Pj1 _Proj1 = {}, _Pj2 _Proj2 = {}) const {
+            // clang-format on
+            if constexpr (same_as<sentinel_t<_Rng1>, unreachable_sentinel_t> //

```

Comment:
Ditto "trying to form meaning here is largely a waste of time".

Reply:
Yeah will udpate some time soon, after thinking about it a bit more